src/web/mjs/connectors/WestManga.mjs

```diff
@@ -1,13 +1,123 @@
-import WordPressMangastream from './templates/WordPressMangastream.mjs';
-
-export default class WestManga extends WordPressMangastream {
+import Connector from "../engine/Connector.mjs";
+import Manga from "../engine/Manga.mjs";
 
+export default class WestManga extends Connector {
     constructor() {
         super();
-        super.id = 'westmanga';
-        super.label = 'WestManga';
-        this.tags = [ 'manga', 'indonesian' ];
-        this.url = 'https://westmanga.me';
-        this.path = '/manga/list-mode/';
+        super.id = "westmanga";
+        super.label = "WestManga";
+        this.tags = ["manga", "manhua", "manhwa", "indonesian"];
+        this.url = "https://westmanga.me";
+        this.queryMangaTitle = "h1";
+
+        // API configuration
+        this.api = {
+            url: "https://data.westmanga.me/api/",
+            nonce: "wm-api-request",
+            accessKey: "WM_WEB_FRONT_END",
+            secretKey: "xxxoidj",
+        };
+    }
+
+    cleanTitle(title) {
+        return title.replace(/bahasa indonesia/i, "").trim();
+    }
+
+    async _getMangas() {
+        const mangaList = [];
+        for (let page = 1; ; page++) {
+            const mangas = await this.getMangasFromPage(page);
+            if (mangas.length === 0) break;
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async getMangasFromPage(page) {
+        const { data } = await this.fetchAPI(`./contents?page=${page}`);
+        return data
+            ? data.map(({ slug, title }) => ({
+                id: slug,
+                title: this.cleanTitle(title),
+            }))
+            : [];
+    }
+
+    async _getChapters(manga) {
+        const {
+            data: { chapters },
+        } = await this.fetchAPI(`./comic/${manga.id}`);
+        return chapters.map(({ slug, number }) => {
+            let title = number.toString().trim();
+            if (/^\d+(\.\d+)?$/.test(title)) {
+                title = `Chapter ${title}`;
+            }
+
+            return {
+                id: slug,
+                title: title,
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        const {
+            data: { images },
+        } = await this.fetchAPI(`./v/${chapter.id}`);
+
+        return images.map((img) => {
+            let url = typeof img === "string" ? img : img.url || img.source;
```

Comment:
What is that? is there something i missed in haruneko? 
Can you provide sample url?